File: SYP.Web/Modules/ServerTypes/Catalog/BrandsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BrandsRow } from "./BrandsRow";

export interface BrandsColumns {
    Id: Column<BrandsRow>;
    Name: Column<BrandsRow>;
    Description: Column<BrandsRow>;
    Logo: Column<BrandsRow>;
    IsActive: Column<BrandsRow>;
    InsertDate: Column<BrandsRow>;
    InsertUserId: Column<BrandsRow>;
    UpdateDate: Column<BrandsRow>;
    UpdateUserId: Column<BrandsRow>;
}

export class BrandsColumns extends ColumnsBase<BrandsRow> {
    static readonly columnsKey = 'Catalog.Brands';
    static readonly Fields = fieldsProxy<BrandsColumns>();
}

File: SYP.Web/Modules/Catalog/Brands/BrandsColumns.cs
namespace SYP.Catalog.Columns;

[ColumnsScript("Catalog.Brands")]
[BasedOnRow(typeof(BrandsRow), CheckNames = true)]
public class BrandsColumns
{
    [EditLink, Width(100)]
    public string Code { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(80)]
    public string Logo { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }

    [Width(80)]
    public int SortOrder { get; set; }
}
